### ISSUE TITLE
:bug: `13.0.4-canary.3` can't find `page.tsx` in app dir with `next dev`

### ISSUE THREAD

**Issue by Chia1104** [vercel/next.js#43019]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

 - Operating System:        
1. Platform: win32        
1. Arch: x64              
1. Version: Windows 10 Pro
 - Binaries:
1. Node: 16.15.1
1. npm: N/A
1. Yarn: N/A
1. pnpm: N/A
 - Relevant packages:
1. next: 13.0.4-canary.3
1. eslint-config-next: N/A
1. react: 18.2.0
1. react-dom: 18.2.0

### What browser are you using? (if relevant)

not relevant

### How are you deploying your application? (if relevant)

not relevant

### Describe the Bug

`13.0.4-canary.3` work incorrectly with `next dev`, because `index.tsx` just be found in `pages` folder.
But next do check the conflict if I put the `page.tsx` in the root of `app` folder.
However  `page.tsx` still can't be found, it will return 404 page in the browser.

> `pages/index.tsx` with `next dev`
> ![13.0.4-canary.3-bug](https://raw.githubusercontent.com/Chia1104/image/main/nextjs/nextjs-bug.PNG)

> `app/page.tsx` or `app/(main)/page.tsx` with `next dev`
> ![13.0.4-canary.3-bug2](https://raw.githubusercontent.com/Chia1104/image/main/nextjs/nextjs-bug2.PNG)

But `appDir` work fine in production build on vercel

### Expected Behavior

The route system should work correctly in `13.0.4-canary.3`

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code will be addressed faster

https://github.com/Chia1104/with-next-13.0.4-canary.3

### To Reproduce

given above

**Comment by BWBama85** [comment]
Same

**Comment by ijjk** [comment]
Hi, this has been updated in `v13.0.4-canary.4`, please update and give it a try!

**Comment by BWBama85** [comment]
> Hi, this has been updated in `v13.0.4-canary.4`, please update and give it a try!

Still 404s for me on all pages. 

```
    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 18.12.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.0.4-canary.4
      eslint-config-next: 13.0.3
      react: 18.2.0
      react-dom: 18.2.0
```

**Comment by ijjk** [comment]
Ah I see thanks for confirm, corrected in `v13.0.4-canary.5` of Next.js
